File: Frontend/Web/donate-spa/src/app/models/Charity.ts
import { Guid } from 'guid-typescript';
import { IModels } from '../shared/crud-service/IDataSource';

export class Charity {
    id: number;
    charityIdentifier: Guid;
    charityName: string;
    contactPerson: string;
    contactNumber: string;
    emailAddress: string;
    updatedAt: Date;
}

export class Charities implements IModels<Charity> {
    offset: number;
    count: number;
    total: number;
    items: Charity[];
}

File: Backend/Donate/Donate/CharityService/Donate.CharityService.API/Data/Entities/Charity.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Donate.Shared.Data;

namespace Donate.CharityService.API.Data.Entities
{
    public class Charity : IDeletableEntity, IAuditableEntity
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Required]
        public Guid CharityIdentifier { get; set; }

        [Required]
        [StringLength(500)]
        public string CharityName { get; set; }

        [Required]
        [StringLength(500)]
        public string ContactPerson { get; set; }

        [Required]
        [StringLength(20)]
        public string ContactNumber { get; set; }

        [Required]
        [StringLength(250)]
        public string EmailAddress { get; set; }

        [Required]
        public bool IsDeleted { get; set; }

        [Required]
        public DateTime UpdatedAt { get; set; }

        [Required]
        [StringLength(250)]
        public string UpdatedBy { get; set; }
    }
}
